src/components/obras/ObraForm.jsx

```diff
@@ -0,0 +1,175 @@
+    isLoading: false,
+  }
+
+  constructor(props) {
+    super(props);
+
+    // If component recieves obra as prop we merge it with initial state (used for editing)
+    this.state = { ...this.state, ...props.obra };
+
+  }
+
+  addObra() {
+    const { title, year, artista, materials, dimensions, medium } = this.state;
+
+    this.setState({ isLoading: true })
+
+    this.props.firebase
+      .push('obras', {
+        title,
+        year,
+        artista,
+        materials,
+        dimensions,
+        medium,
+      })
+      .then(() => {
+        this.setState({ isLoading: false })
+        this.props.history.push('/obras');
+      })
+
+  }
+
+  updateObra() {
+    const { title, year, artista, materials, dimensions, medium } = this.state;
+
+    this.setState({ isLoading: true })
+
+    this.props.firebase
+      .update(`obras/${this.props.id}`, {
+        title,
+        year,
+        artista,
+        materials,
+        dimensions,
+        medium,
+      })
+      .then(() => {
+        this.setState({ isLoading: false })
+      })
+
+  }
+
+  render() {
+    return (
+      <form onSubmit={event => event.preventDefault()}>
+        <div className='grid-row margin-bottom-basic justify-end'>
+          <div className='grid-item'>
+            <button className='button' onClick={() => this.props.id ? this.updateObra() : this.addObra()}>
+              Guardar{ this.props.id ? '' : ' Nuevo'}
+            </button>
+          </div>
+        </div>
+
+        <div className='grid-row margin-bottom-basic'>
+          <div className='grid-item item-s-12'>
+            <h4 className='font-size-small font-bold margin-bottom-tiny'><label htmlFor='title'>Title</label></h4>
+            <input
+              id='title'
+              name='title'
+              type='text'
+              disabled={this.state.isLoading}
+              value={this.state.title}
+              onChange={ event => this.setState({ title: event.target.value })}
+            />
+          </div>
+        </div>
+
+        <div className='grid-row margin-bottom-basic'>
+          <div className='grid-item item-s-12'>
+            <h4 className='font-size-small font-bold margin-bottom-tiny'><label htmlFor='year'>Year</label></h4>
+            <div className='grid-row align-items-center'>
+              <input
+                id='year'
+                name='year'
+                type='text'
+                disabled={this.state.isLoading}
+                value={this.state.year}
+                onChange={ event => this.setState({ year: event.target.value })}
+              />
+            </div>
+          </div>
+        </div>
+
+        <div className='grid-row margin-bottom-basic'>
+          <div className='grid-item item-s-12'>
+            <h4 className='font-size-small font-bold margin-bottom-tiny'><label htmlFor='artista'>Artista</label></h4>
+            <input
+              id='artista'
+              name='artista'
+              type='text'
```

Comment:
this should be a select field filled with artists: id / name as key / value pair

Reply:
yh it says in the PR artist select is outstanding

Reply:
oic